cli/cloud/runner/multifile_orchestrator.go

```diff
@@ -209,27 +210,58 @@ func (o orchestrator) Run(ctx context.Context, opts RunOptions, outputFormat str
 	return exitCode, nil
 }
 
-func (o orchestrator) getDefinitionFiles(file []string) ([]string, error) {
-	files := make([]string, 0)
-
+func (o orchestrator) getDefinitionFiles(file []string, resourceName string) (files []string, err error) {
 	for _, f := range file {
 		files = append(files, fileutil.ReadDirFileNames(f)...)
 	}
 
+	if resourceName != "" {
+		o.logger.Debug("filtering files by resource name", zap.String("resourceName", resourceName))
+		files, err = o.filterFilesByResourceName(files, resourceName)
+		if err != nil {
+			return nil, fmt.Errorf("cannot filter files by resource name: %w", err)
+		}
+	}
+
+	o.logger.Debug("selected files", zap.Strings("files", files))
+
 	return files, nil
 }
 
+func (o orchestrator) filterFilesByResourceName(files []string, resourceName string) ([]string, error) {
+	filteredFiles := make([]string, 0)
+
+	for _, f := range files {
+		o.logger.Debug("reading file", zap.String("file", f))
+		file, err := fileutil.Read(f)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read file: %w", err)
+		}
+
+		// EqualFold means compare case insensitive
+		if ft := file.Type(); !strings.EqualFold(ft, resourceName) {
+			o.logger.Debug("skipping file", zap.String("type", ft), zap.String("file", f))
+			continue
+		}
+
+		o.logger.Debug("file matches", zap.String("file", f))
+		filteredFiles = append(filteredFiles, f)
+	}
+
+	return filteredFiles, nil
+}
+
 func (o orchestrator) runByFiles(ctx context.Context, opts RunOptions, resourceFetcher runner.ResourceFetcher, vars *varset.VarSets, varsID string, runGroupID string) ([]any, []runner.RunResult, error) {
 	resources := make([]any, 0)
 	runsResults := make([]runner.RunResult, 0)
 	var mainErr error
 
-	hasDefinitionFilesDefined := opts.DefinitionFiles != nil && len(opts.DefinitionFiles) > 0
+	hasDefinitionFilesDefined := len(opts.DefinitionFiles) > 0
```

Comment:
this is not related, but it was causing an annoying warning

![Screenshot 2024-09-17 at 13 34 15](https://github.com/user-attachments/assets/b2cc52f2-8445-4740-a453-ef777c3df71d)